In [1]:
import pandas as pd
import numpy as np


In [10]:
passanger.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


In [11]:
passanger = pd.read_csv('train.csv')
passanger.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [12]:
print("Original dataset contain shape: ")
passanger.shape

Original dataset contain shape: 


(891, 12)

### 1 Dealing with missing values
- 1.1 Drop the missing values.


In [13]:
removed_row = passanger.dropna(axis=0,how='any',thresh=None,subset=None,inplace=False)
print("After removing the missing value shape of dataset")
removed_row.shape

After removing the missing value shape of dataset


(183, 12)

#### Observation: 
- Now titanic dataset hugly reomved row due to missing value
- Now there is only 183 row and 12 columns

In [14]:
removed_row.head()

PassengerId  Survived  Pclass  \
1             2         1       1   
3             4         1       1   
6             7         0       1   
10           11         1       3   
11           12         1       1   

                                                 Name     Sex   Age  SibSp  \
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
10                    Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   

    Parch    Ticket     Fare Cabin Embarked  
1       0  PC 17599  71.2833   C85        C  
3       0    113803  53.1000  C123        S  
6       0     17463  51.8625   E46        S  
10      1   PP 9549  16.7000    G6        S  
11      0    113783  26.5500  C103        S

In [18]:
passanger.dropna(axis=1).head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    male      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female      1      0   
2                             Heikkinen, Miss. Laina  female      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female      1      0   
4                           Allen, Mr. William Henry    male      0      0   

             Ticket     Fare  
0         A/5 21171   7.2500  
1          PC 17599  71.2833  
2  STON/O2. 3101282   7.9250  
3            113803  53.1000  
4            373450   8.0500

In [20]:
passanger['Age'].fillna(value= passanger['Age'].median()).head()


0    22.0
1    38.0
2    26.0
3    35.0
4    35.0
Name: Age, dtype: float64

### 2 The apply function

In [24]:
def get_full_city_name(cityCode):
    if (cityCode == "S"):
        return "Southampton"
    elif (cityCode == "C"):
        return "Cherbourg"
    elif (cityCode == "Q"):
        return "Queenstown"
passanger['Embarked'].apply(get_full_city_name).head()


0    Southampton
1      Cherbourg
2    Southampton
3    Southampton
4    Southampton
Name: Embarked, dtype: object

In [26]:
# Updating the fares for all the passengers
passanger['Fare'].apply(lambda fare: 100*fare).head()


0     725.00
1    7128.33
2     792.50
3    5310.00
4     805.00
Name: Fare, dtype: float64

In [27]:
# Find the max fare of traveller
passanger['Fare'].apply(lambda fare: 100*fare).max()

51232.919999999998